### ISSUE TITLE
Global `history.push` suppression and always send `WebViewMessageTypes.NAVIGATION` event when `window.ReactNativeWebView` is `true`

### ISSUE THREAD

**Issue by lorenjohnson** [Hylozoic/hylo-evo#1326]
This will be a replacement for the current `hyloApp` layout flag pattern. It should allow some/all of the custom logic within components here on Web (i.e. within MapExplorer, etc) to be removed. 

This new flag is being sent as of the upcoming release of HyloApp, so there should be no changes on mobile.

This change will fix at least the following issues in the App:

* All Topics WebView doesn't navigate out to selected Topics but stays within the WebView and on HyloWeb
* UserSettings > Invites & Requests redirect loop when clicking on a group
* Saved Searches for Map, confirm fixed after navigation suppression is added in HyloWeb

 Basically following this advice: https://medium.com/@tanjohnny/webview-of-your-spa-in-react-native-a06b0aa2e85d
